modmesh/app/euler1d.py

```diff
@@ -142,215 +301,72 @@ def __init__(self, shocktube):
         self.checkbox_select_num = 3
-
-    @Slot(bool)
-    def save_all(self, checked=False):
+        self.plot_holder = State()
+        self.set_solver_config()
+        self.setup_timer()
+        self.plot_holder.plot = self.build_single_figure()
+
+    def init_solver(self, gamma=1.4, pressure_left=1.0, density_left=1.0,
+                    pressure_right=0.1, density_right=0.125, xmin=-10,
+                    xmax=10, ncoord=201, time_increment=0.05):
         """
-        This callback function don't care button's checked state,
-        therefore the checked state is not used in this function.
+        This function is used to initialize shock tube solver and setup
+        the initial conition.
 
-        :param checked: button is checked or not
-        :return: nothing
+        :return: None
         """
-        fig = QPixmap(self.figure_container.size())
-        self.figure_container.render(fig)
-
-        options = QFileDialog.Options()
-        options |= QFileDialog.DontUseNativeDialog
-        fileName, _ = QFileDialog.getSaveFileName(None, "Save file", "",
-                                                  "All Files (*)",
-                                                  options=options)
-
-        if fileName != "":
-            fig.save(fileName, "JPG", 100)
+        self.st = euler1d.ShockTube()
+        self.st.build_constant(gamma, pressure_left, density_left,
+                               pressure_right, density_right)
+        self.st.build_numerical(xmin, xmax, ncoord, time_increment)
+        self.st.build_field(t=0)
 
-    def _update_layout(self):
-        while self.main_layout is not None and self.main_layout.count():
-            item = self.main_layout.takeAt(0)
-            if isinstance(item, QLayout):
-                self.delete_layout(item)
-            else:
-                item.widget().setParent(None)
-                item.widget().deleteLater()
-
-        self.main_layout.deleteLater()
-        self.main_container.update()
-
-        @Slot(QObject)
-        def del_cb(obj=None):
-            """
-            Qt objects are managed by Qt internal event loop,
-            hence it will not be deleted immediately when we called
-            deleteLater() and a QWidget only accept one layout at
-            the same time, therefore a object delete callback is
-            needed to set new layout after old layout deleted.
-
-            This callback will be called before the obj destroyed,
-            in this callback only update the layout, therefore the obj
-            is not used in this callback.
-
-            :param obj: The object about to be destroyed
-            :return: nothing
-            """
-            if self.use_grid_layout:
-                self.build_lines_grid_layout()
-            else:
-                self.build_lines_single_plot()
-
-        self.main_layout.destroyed.connect(del_cb)
-
-        self.main_layout.update()
-        self.main_container.update()
-
```

Comment:
Removing the GUI layout update callback. Once the GUI needs to be refreshed, this part will be handled in the PUI DOM update.